meson.build

```diff
@@ -432,10 +433,29 @@ if build_tools
         dependencies: dep_libxkbcommon,
     )
 
+    # Tool: xkbcli
     executable('xkbcli', 'tools/xkbcli.c',
                dependencies: tools_dep, install: true)
     install_man('tools/xkbcli.1')
 
+    bash_completion_path = get_option('bash-completion-path')
+    if bash_completion_path == ''
+        bash_completion = dependency('bash-completion', required : false)
+        if bash_completion.found()
+            bash_completion_path = bash_completion.get_variable(pkgconfig : 'completionsdir')
+        else
+            bash_completion_path = get_option('datadir') / 'bash-completion/completions'
+        endif
+    endif
+    if bash_completion_path != 'no'
+        install_data(
+            'tools/xkbcli-completion.sh',
+            rename: 'xkbcli',
+            install_dir: bash_completion_path
+        )
+    endif
```

Comment:
handwavy: I'm wondering if it's better to have a `bash-completion` as meson feature and the path as a separate meson option that is either `''` or an explicit path. Mostly because while I can see the point of disabling bash completion, setting the path is probably quite a specific niche case. And setting a path string as `no` feels like an old autotools leftover, meson options cost very little in comparison.

This is a suggestion only, feel free to ignore.

Reply:
I agree. I adapted the meson script from `systemd` as Ran proposed and we can continue improving on it.